scripts/core/editor3/components/HighlightsPopup.jsx

```diff
@@ -123,34 +79,31 @@ export class HighlightsPopup extends Component {
     component() {
         const {store} = this.context;
         const position = this.position();
-
-        var highlightsAndSuggestions = [];
-
-        var suggestionEntityKey = this.getSelectedSuggestionEntityKey();
-
-        if (suggestionEntityKey !== null) {
-            var suggestionEntity = Entity.get(suggestionEntityKey);
-            const {selection, suggestionText} = this.getRangeAndTextForEntityKey(
-                this.props.editorState, suggestionEntityKey
-            );
-
-            highlightsAndSuggestions = [
-                ...highlightsAndSuggestions,
-                {type: suggestionEntity.getType(), value: {...suggestionEntity.toJS(), suggestionText, selection}}
-            ];
-        }
+        const types = ['DELETE_SUGGESTION', 'ADD_SUGGESTION'];
+        const suggestionStyle = getHighlightStyle(this.props.editorState, types);
+        let highlightsAndSuggestions = [];
 
         if (this.styleBasedHighlightsExist()) {
             this.getInlineStyleForCollapsedSelection()
                 .filter(this.props.highlightsManager.styleNameBelongsToHighlight)
                 .forEach((styleName) => {
                     const highlightType = this.props.highlightsManager.getHighlightTypeFromStyleName(styleName);
+                    let data = this.props.highlightsManager.getHighlightData(styleName);
+
+                    if (types.indexOf(highlightType) !== -1) {
+                        const {selection, suggestionText} = getRangeAndTextForStyle(
+                            this.props.editorState, suggestionStyle
+                        );
+
+                        data = {...data, suggestionText, selection};
```

Comment:
The selection is used for accept and reject operations. The suggestion widget text and selection is refreshed on every change.

Reply:
Wouldn't it be better to use `highlightId`(inline style name) for accepting/rejecting suggestions? When you get a `highlightId` you can then dive into the `editorState` and get the selection for that `highlihgtId`. Just a suggestion, I'm not sure if it would work out this way.

Reply:
Done